File: sdks/node/aiProcessor.js
const fetch = require('node-fetch');

class AIProcessor {
  constructor(baseUrl, apiKey){
    this.baseUrl = baseUrl.replace(/\/+$/, "");
    this.headers = { "Authorization": apiKey, "Content-Type": "application/json" };
  }

  async process({template=null, description=null, data=null, response_schema=null, options=null} = {}){
    const res = await fetch(`${this.baseUrl}/v1/process`, {
      method: "POST",
      headers: this.headers,
      body: JSON.stringify({template, description, data, response_schema, options})
    });
    if (!res.ok) throw new Error(await res.text());
    return res.json();
  }

  async getJob(jobId){
    const res = await fetch(`${this.baseUrl}/v1/jobs/${jobId}`, { headers: this.headers });
    if (!res.ok) throw new Error(await res.text());
    return res.json();
  }
}

module.exports = AIProcessor;

File: sdks/python/ai_processor.py
import requests
import json

class AIProcessorClient:
    def __init__(self, base_url: str, api_key: str):
        self.base_url = base_url.rstrip("/")
        self.headers = {"Authorization": api_key, "Content-Type": "application/json"}

    def process(self, template=None, description=None, data=None, response_schema=None, options=None):
        payload = {
            "template": template,
            "description": description,
            "data": data,
            "response_schema": response_schema,
            "options": options
        }
        r = requests.post(f"{self.base_url}/v1/process", headers=self.headers, json=payload, timeout=60)
        r.raise_for_status()
        return r.json()

    def get_job(self, job_id):
        r = requests.get(f"{self.base_url}/v1/jobs/{job_id}", headers=self.headers, timeout=30)
        r.raise_for_status()
        return r.json()
